buildenv/jenkins/jobs/infrastructure/signedOffByCheck

```diff
@@ -0,0 +1,82 @@
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+def FAIL = false
+def SRC_REPO = 'https://github.com/${ghprbGhRepository}.git'
+def BAD_FILES = []
+def HASHES = '###################################'
+def WHITELIST_FILES = []
+def REGEXLIST = []
+def COMMITS = []
+def white = []
+def BAD_COMMITS = []
+
+timeout(time: 6, unit: 'HOURS') {
+    stage('Copyright Check') {
+        node ('worker') {
+            timestamps {
+                git url: SRC_REPO
+                sh "git fetch --tags --progress origin +refs/pull/*:refs/remotes/origin/pr/*"
+                sh "git checkout --detach ${ghprbActualCommit}"
+                sh 'git fetch origin'
+                COMMITS = sh(
+                     script : "git log origin/${ghprbTargetBranch}..HEAD",
+                     returnStdout: true
+                     ).trim()
+                echo COMMITS
+                if (COMMITS == "") {
+                    echo "There are no commits to check for signoffby message"
+                }else{
+                    def str = COMMITS.split("(?=commit [a-f0-9]{40}+\n(Merge: [a-f0-9 ]+)*\n*Author: .*\nDate:   .*+\n)");
+                    str.each { string ->
+                        if (string != ""){
+                            def splitString = string.split("\\r?\\n")
+                            def last = splitString.last()
+                            def secondLast = splitString[splitString.size()-2]
+                            def second = splitString[1] 
+                            if (second.contains("Merge")){
+                                echo "ignore Merge Commit: '${string}'" 
+                            }else if (("${last}" ==~ /\s*Signed\-off\-by\:\s*([a-zA-Z]\s*)+\<[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}\>/ ) && secondLast==~/\s*/) {                                 
+                                echo "checking Commit: '${string}' appears to be in the right format"                                                                                      
+                             }else{
+                                echo "FAILURE - Commit message: '${string}' appears to be incorrect"
+                                BAD_COMMITS << "${string}"
+                                FAIL = true
+                             }
+                         }
+                    }
+                          if (FAIL) { 
+                             echo "${HASHES}"
+                             echo "FAILURE - commit appears to be incorrect"
```

Comment:
similar to line 59
`FAILURE - The following commits appear to have an incorrect sign-off`

Reply:
fixed